### DISCUSSION TITLE
Facing issue in Enabling channel through Mirth CLI

### DISCUSSION THREAD

**Discussion by manaschandanroy** [nextgenhealthcare/connect#4936] [Q&A]
I am running the Mirth CLI and somehow the when i list the channels , my channel is showing as disabled. even after running channel enable and listing the channel again it says channel enabled "NO". can anyone please tell me why in mirth-connect 3.11 , the channel is disabled.

Any idea guys ?

**Comment by pacmano1** [comment]
I do not see that behavior on 3.9.1 or 3.12.   I do see it on 3.11.    (see next comment after more discovery)

**Reply by pacmano1** [reply]
Interesting - so if I use the tar.gz (which has mccommand) rather than the mirth installer for 3.11 which does not have mccommand, it works as designed for 3.11.   Is your CLI install standalone?

**Reply by arch2199** [reply]
We are using the official mirth docker image. It provides us with the required mirth-cli-launcher.jar in it. We have also checked it with mccommand, still the imported channel remains to be disabled. We have also tried it by downgrading the mirth version in the Dockerfile.
![image](https://user-images.githubusercontent.com/64971441/147807135-0b2b5ad1-4dff-429c-ad7d-c94de4131d56.png)

**Reply by manaschandanroy** [reply]
@pacmano1 any idea what are we missing and what can be done here ? . Additionally we are not able to download the tar or zip for 3.12 from the official site . It just redirects to another page and then nothing gets downloaded .

**Comment by pacmano1** [comment]
Docker drives me up the wall.  Mostly because people use containers and don't understand how to set up an application from scratch on a non-VM or container.    I get the value of containers of course.   My comment is not directed at you.

What I would say is replicate the bug in a non container then post a bug here in github in the issue section.
 
A command line wget works for the download.

`wget http://downloads.mirthcorp.com/connect/3.12.0.b2650/mirthconnect-3.12.0.b2650-unix.tar.gz`

I don't use the installer because it is stupid and runs things as root, which is very bad practice.   Which is sort of funny really because the Docker image runs mirth as the user "mirth.

**Reply by manaschandanroy** [reply]
@pacmano1 additionally do you recommend that we download the 3.9 tar and install it on a Linux VM and try , asking this coz in your initial comments you said that unzipping the tar and installing it is kinda working in your earlier comments . Should we do that to find a solution to make it work if it does ?

**Reply by pacmano1** [reply]
Yeah - so the tar.gz worked for all versions for me - that one includes mccommand.

**Reply by manaschandanroy** [reply]
@pacmano1 thanks . I & Archana will try the same and see if we have a breakthrough with that

**Reply by arch2199** [reply]
@pacmano1 We have used the tar file version 3.12 as you have suggested. We are able to start the CLI using the mccommand, even with that the issue persists. We are using an official channel xml file (that supports the conversion of patient resource from HL7v2 to HL7-FHIR). We can send you the file on email. Can you verify on your system to see if this channel is getting enabled when you import it?

![image](https://user-images.githubusercontent.com/64971441/147909883-5e59397c-39a7-4680-9909-07553747ef59.png)

**Reply by arch2199** [reply]
@pacmano1 While using the mirthconnect application through console, we have added a bunch of necessary extensions during the initial setup. Our channel is perfectly running and yielding the expected result in the GUI (in Windows machine). 
When we are trying to connect to the CLI, everything is in place but we are not sure about how to run or execute these extensions. Can you help us with how and where to install the required extension files.

List of extensions used for the channel in MirthConnect GUI:
![image (1)](https://user-images.githubusercontent.com/64971441/147939388-1b96dae0-22d4-4988-95a5-e00c43569628.png)

**Reply by pacmano1** [reply]
Firstly, please understand that I do not work for Nextgen so I am not official "support".   The FHIR plugin is not free, it is part of the Gold license pack, so I assume you have a license.   Since you have a license you can open a support ticket with Nextgen.   

I don't understand your question "how to execute extensions".   Most of the extensions you show in your screenshot have their correlated user interface in the Mirth Administrator Tool.    Also I run the CLI with `mccomand`, which is a wrapper for the jar.   I don't know how you are starting the CLI.

**Reply by manaschandanroy** [reply]
@pacmano1 Hi  I am sure Archana just reached out for support with your vast experience as we are still noobies and we really appreciate your support on this.  Her question is primarily focused around the area that custom extension if you want to install in CLI , just putting  a subfolder inside extension folder wont be enough. If you are aware as to how can we install it(Command) so that CLI picks these custom extension too.
We believe the reason our channel is showing not enabled because these supporting extensions are not available. We led to this conclusion as we created an empty channel which is by default showing as enabled

![image](https://user-images.githubusercontent.com/37400576/147944012-71f0cce6-8e19-4ed5-ac09-0b83d8f09a35.png)

**Reply by pacmano1** [reply]
Ah, that make more sense now.  

So I'll have to defer to others on custom extension installation, I only use Nextgen provided extensions. I was under the impression the extension registered itself based on the assorted XML "stuff" in its subdirectory.

**Comment by jonbartels** [comment]
Is the channel deployable with an HTTP API call? 

I think that is important. If the channel is deployable from the Admin GUI and the API call, it isolates the problem to the CLI. If it is not deployable from the API and the CLI then... something really strange and new is wrong.
